### ISSUE TITLE
Unexpected casting on ColumnOptions.transformer

### ISSUE THREAD

**Issue by pueue** [typeorm/typeorm#4137]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

There is a string data type column. And that can have only `'Y'` or `'N'` as a value. Yes. On the legacy database.

```ts
@Column({
  transformer: {
    from: (v: string) => (v === 'Y'),
    to: (v: boolean) => (v ? 'Y' : 'N'),
  }
})
public verified: boolean;
```

I tried to convert this column to boolean on my model like above. But `verified` was always false. Because if a property type was boolean, a value from database is cast to boolean before running `transformer.form` function. So `true === 'Y'` was always false.

I think it related to
https://github.com/typeorm/typeorm/blob/master/src/driver/mysql/MysqlDriver.ts#L475

**Comment by Kononnable** [comment]
Yes, it looks like related to the mentioned line. However we can't just remove it - in some cases people might store boolean as `0,1` etc. I think we have to figure out how to deal with it, probably transformers aren't enough or we might have to add a new kind of transformers - which would take place before/instead of those lines.

**Comment by vlapo** [comment]
We should run `from` transformer on begining of `prepareHydratedValue` method. Or maybe pass second argument with raw value from database. Something like:
```ts
transformer: {
    from: (hydratedValue: string, rawValue: any) => (rawValue === 'Y'),
    to: (v: boolean) => (v ? 'Y' : 'N'),
  }
```

**Comment by Kononnable** [comment]
Running before `prepareHydratedValue` would probably broke current transformers and will force them to hydrate value on their own(not just convert it). Option with second parameter looks promising. It would be a breaking change, but a minor one.

**Comment by vlapo** [comment]
Why is second parameter breaking change? If the first parameter of transformer is still `hydratedValue` as before it is just adding new parameter. It will not break any existing functionality. Or am I missing something ? :)

**Comment by Kononnable** [comment]
After that change you would have to change every user-code transformer to match function definition. So such change would break compilation of user code(or at least I think it will).

**Comment by vlapo** [comment]
Ou sorry. I thought we have optional parameter in from transformer. But we can define second parameter as optional if we want to have this change available before 0.3.0

**Comment by Kononnable** [comment]
Right, forgot that we can add an optional parameter and transformers with and without new parameter will be compatible.

**Comment by kevinsproles** [comment]
Have the same issue. Boolean for api, but 'Y' or 'N' as a value in db on the legacy database.

Was this issue solved somehow?
